packages/form-field/src/index.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react'
+import * as styled from './styles'
+import {useDefaults} from '@smashing/theme'
+import {useFormContext} from '@smashing/form'
+import '@smashing/theme'
+import {
+  FormFieldProps,
+  FormFieldLabelAppearance,
+  FormFieldAlertAppearance
+} from './types'
+
+const INVALID_CONTEXT_MESSAGE = `[@smashing/form-field] FormField can be used only in @smashing/form context:
+const {Form} = useForm({/* Your form config */})
+
+return (
+  <Form>
+    <FormField component={Input} name="email" />
+  </Form>
+)
+`
+
+export const FormField: React.FC<FormFieldProps> = ({
+  component,
+  description,
+  hint,
+  htmlFor,
+  id,
+  label,
+  name,
+  className,
+  ...props
+}) => {
+  const {Field, ErrorMessage} = useFormContext()
+  const {labelAppearance, alertAppearance, ...inputProps} = useDefaults(
+    'formField',
+    props,
+    {
+      labelAppearance: 'block' as FormFieldLabelAppearance,
+      alertAppearance: 'block' as FormFieldAlertAppearance
+    }
+  )
+
+  if (!ErrorMessage || !Field) {
+    console.error(INVALID_CONTEXT_MESSAGE)
+    return null
+  }
+
+  return (
+    <styled.FormField labelAppearance={labelAppearance} className={className}>
+      <styled.Label
+        appearance={labelAppearance}
```

Comment:
It should be possible to provide label's variant/size on either label, form or theme level

Reply:
Added `labelVariant`